lib/ami_spec/aws_instance.rb

```diff
@@ -37,6 +37,11 @@ def start
       tag_instance
     end
 
+    def terminate
+      @instance.terminate
+      @instance.wait_until_terminated
```

Comment:
To successfully delete a security group, any EC2 instances using it must be terminated. Hence we'll need to start waiting for this EC2 instance fully terminate before attempting to delete the security group.